backend/tests/integration/test_b07_p8_topology_closure_pack.py

```
@@ -0,0 +1,498 @@
+        path = _repo_root() / "artifacts" / "b07-p8"
+    path.mkdir(parents=True, exist_ok=True)
+    return path
+
+
+def _runtime_sync_db_url() -> str:
+    explicit = os.getenv("B07_P8_RUNTIME_DATABASE_URL")
+    if explicit:
+        return explicit
+    runtime = os.getenv("DATABASE_URL")
+    if not runtime:
+        raise RuntimeError("DATABASE_URL or B07_P8_RUNTIME_DATABASE_URL is required")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)